src/oasis-nexus/generated/api.ts

```diff
@@ -1555,10 +1562,28 @@ export const ConsensusEventType = {
 export interface TxError {
   /** The status code of a failed transaction. */
   code: number;
-  /** The message of a failed transaction. */
+  /** The message of a failed transaction.
+This field, like `code` and `module`, can represent an error that originated
+anywhere in the paratime, i.e. either inside or outside a smart contract.
+
+A common special case worth calling out: When the paratime is 
+EVM-compatible (e.g. Emerald or Sapphire) and the error originates 
+inside a smart contract (using `revert` in solidity), the following 
+will be true:
+- `module` will be "evm" and `code` will be 8; see [here](https://github.com/oasisprotocol/oasis-sdk/blob/runtime-sdk/v0.8.3/runtime-sdk/modules/evm/src/lib.rs#L128) for other possible errors in the `evm` module.
+- `message` will contain the best-effort human-readable revert reason.
+ */
   message?: string;
   /** The module of a failed transaction. */
   module?: string;
+  /** The error parameters, as decoded using the contract abi. Present only when
+- the error originated from within a smart contract (e.g. via `revert` in Solidity), and 
+- the contract is verified or the revert reason is a plain String.
+If this field is present, `message` will include the name of the error, e.g. 'InsufficentBalance'.
+Note that users should be cautious when evaluating error data since the
+data origin is not tracked and error information can be faked.
+ */
+  revert_params?: EvmAbiParam[];
```

Comment:
Diff doesn't make it clear what all the new fields are. We should extract a list of new fields to determine which ones we should start using in existing code

Reply:
Split in https://github.com/oasisprotocol/explorer/pull/1190/commits/cd0bf54c7b5f7441cda2cdb03be1161e7e290d9d

Reply:
Split in https://github.com/oasisprotocol/explorer/pull/1190/commits/86ff7f729ec986388df07f76cb97132273751f00

Reply:
https://github.com/oasisprotocol/explorer/issues/1222